### PR TITLE
[XM] Xcode8b5 AppKit API changes

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

**Comment by spouliot** [comment]
Bots are not yet updated (in progress by RE) so the results will be inconclusive.

**Comment by chamons** [comment]
@spouliot Just getting it out there for person review. I'll re-build once RE gets bots happy.

**Comment by xamarin-release-manager** [comment]
Build success

File: src/appkit.cs [hunk 1]
```diff
@@ -23562,14 +23566,14 @@ interface NSFilePromiseProvider : NSPasteboardWriting
 	interface NSFilePromiseProviderDelegate
```
Comment (spouliot) [inline src/appkit.cs]: if the two changes here are not breaking changes then it means it's a new type in 10.12 and there should be an availability attribute on the type

it also should be marked as a `[Protocol]` and `[Model]` and anything that use it should do so from the interface that represent the protocol (and not the concrete type that the model provides)

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by spouliot** [comment]
it _should_ need to be rebased on the head of `xcode8` to get the Xcode bump

otoh it's not executed on a 10.12 bot (on jenkins) so the results won't change (and they were ok on 10.11 earlier).

You still need to fix my comments above, but it can be committed afterward

File: src/appkit.cs (hunk 1 above)
Comment (chamons) [inline src/appkit.cs]: This is a github diff but apparently (look at the line numbers even if you expand the diff).

The real code is:

	[Mac (10,12)]
	[Protocol]
	interface NSFilePromiseProviderDelegate
	{

**Pull request merged**